### ISSUE TITLE
[Bug] Double data axis, precision of x-axis  is lost

### ISSUE THREAD

**Issue by yyk112233** [apache/echarts#19282]
### Version

5.4.3

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?c=line-in-cartesian-coordinate-system

### Steps to Reproduce

![image](https://github.com/apache/echarts/assets/150311073/ce5922b5-52eb-49d7-bf40-bd8c1105826e)

https://echarts.apache.org/examples/zh/editor.html?c=line-in-cartesian-coordinate-system

### Current Behavior

some points are overlapping

### Expected Behavior

![image](https://github.com/apache/echarts/assets/150311073/1679f8af-4790-43fb-a0e1-c1bfd0d467b7)

### Environment

```markdown
- OS:  macOS Monterey
- Browser:   Chrome
- Framework:
```

**Comment by helgasoft** [comment]
``` xAxis: {type:'category'}```

NB: please close issue if problem solved.

**Comment by yyk112233** [comment]
If I want to continue using the data axis, is there a way around this?

**Comment by helgasoft** [comment]
nope, it's either _value_ or _category_ type, nothing else under the sun.

**Comment by yyk112233** [comment]
![image](https://github.com/apache/echarts/assets/150311073/bae64d6f-32b4-420c-a361-294dc46e4506)
use value type in x-axis will lose precision, i want to preserve precision, is there a way around this?

**Comment by plainheart** [comment]
Maybe you can customize the tooltip using `tooltip.formatter` to circumvent the precision loss issue.
